polling/nf_configuration_test.go

```diff
@@ -0,0 +1,290 @@
+// SPDX-FileCopyrightText: 2025 Canonical Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+/*
+ * NF Polling Unit Tests
+ *
+ */
+
+package polling
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/omec-project/amf/factory"
+	"github.com/omec-project/openapi/models"
+	"github.com/omec-project/openapi/nfConfigApi"
+)
+
+func makeAccessMobilityConfig(mcc, mnc, sst string, sd string, tacs []string) nfConfigApi.AccessAndMobility {
+	sstUint64, err := strconv.ParseUint(sst, 10, 8)
+	if err != nil {
+		panic("invalid SST value: " + sst)
+	}
+	sstint := int32(sstUint64)
+	return nfConfigApi.AccessAndMobility{
+		PlmnId: nfConfigApi.PlmnId{
+			Mcc: mcc,
+			Mnc: mnc,
+		},
+		Snssai: nfConfigApi.Snssai{
+			Sst: sstint,
+			Sd:  &sd,
+		},
+		Tacs: tacs,
+	}
+}
```

Comment:
What about returning an error instead of panic here and then check is `err != nil` to call t.Fatalf()`
```suggestion
func makeAccessMobilityConfig(mcc, mnc, sst string, sd string, tacs []string) (nfConfigApi.AccessAndMobility, error) {
	sstUint64, err := strconv.ParseUint(sst, 10, 8)
	if err != nil {
		return nfConfigApi.AccessAndMobility{}, err
	}
	sstint := int32(sstUint64)
	return nfConfigApi.AccessAndMobility{
		PlmnId: nfConfigApi.PlmnId{
			Mcc: mcc,
			Mnc: mnc,
		},
		Snssai: nfConfigApi.Snssai{
			Sst: sstint,
			Sd:  &sd,
		},
		Tacs: tacs,
	}, nil
}
```

```go
	accessMobility1, err := makeAccessMobilityConfig("222", "02", "1", "1", []string{"1"})
	if err != nil {
		t.Fatalf("Failed to create access mobility config: %v", err)
	}
```